qiskit_nature/second_q/operators/fermionic_op.py

```diff
@@ -161,6 +163,63 @@ def _validate_keys(cls, keys: Collection[str], register_length: int) -> None:
                         f"{register_length}."
                     )
 
+    @classmethod
+    def _validate_polynomial_tensor_key(cls, keys: Collection[str]) -> None:
+        allowed_chars = {"+", "-"}
+
+        for key in keys:
+            if set(key) - allowed_chars:
+                raise QiskitNatureError("TODO.")
+
+    @classmethod
+    def from_polynomial_tensor(cls, tensor: PolynomialTensor) -> FermionicOp:
+        """TODO."""
+        cls._validate_polynomial_tensor_key(tensor.keys())
+
+        data: dict[str, complex] = {}
+
+        for key in tensor:
+            if key == "":
+                # TODO: deal with complexity
+                data[""] = cast(float, tensor[key])
+                continue
+
+            label_template = " ".join(f"{op}_{{}}" for op in key)
+
+            # PERF: this matrix unpacking is a performance bottleneck
+            # I am including multiple solutions below to start a discussion on how to implement
+            # this. All of these are single-threaded right now...
+
+            # Another aspect would be to filter non-zero terms. However, so far all solutions I have
+            # tried result in significant performance penalties.
+
+            # nditer = np.nditer(tensor[key], flags=["multi_index"], op_flags=["readonly"])
+            # for coeff in nditer:
+            #     data[label_template.format(*nditer.multi_index)] = coeff
+
+            # An alternative solution, not based on a sequential numpy iterator. Could we use this
+            # to our advantage and parallelize this to some degree? A naive attempt at using
+            # qiskit.tools.parallel_map hurt more, than it helped...
+            mat = cast(np.ndarray, tensor[key])
+            for index in product(range(tensor.register_length), repeat=len(key)):
+                # TODO: deal with complexity
+                data[label_template.format(*index)] = cast(float, mat[index])
+
+            # and yet another solution which uses the same numpy iterator as above, but the C-API
+            # instead of Pythonic iterator, which according to their docs could be beneficial when
+            # requiring access to the iterator coordinates/indices
+            # however, here it is even slower than the variant above
+            # Reference: https://numpy.org/doc/stable/reference/generated/numpy.nditer.html
+            # nditer = np.nditer(tensor[key], flags=["multi_index"], op_flags=["readonly"])
+            # while not nditer.finished:
+            #     data[label_template.format(*nditer.multi_index)] = nditer[0]
+            #     nditer.iternext()
```

Comment:
I would like to discuss these various solutions before settling on one

Reply:
Yes indeed. I think there is another comment below the section I selected here which mentions that, once `sparse` is supported, the logic above will need to be adapted to also support that setting